### PR TITLE
Sandy and Isa - Add new item

### PR DESCRIPTION
**Pull request by isaabutaa**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

- Changed `AddItem` component and added a form with an input for the item and frequency for next purchase using radio buttons
- Added `item`, `frequency`, and `lastPurchasedDate` saved state values
- Created `handleChange` function to save user input values (item name & purchase frequency) to state
- Used `getToken()` in a `useEffect()` to set token inside localStorage, but we will have to change that after Kelsey and Hulya's PR gets approved and we merge our branches together
- Attached `name`, `frequency`, `token`, and `lastPurchasedDate` to the item template and added it to the firestore database 
- Added an alert that gives feedback to user indicating that the item was added to their list. After the alert, the user is redirected to the item list view
- The list view has been changed to render an unordered list of items, showing the item names 
- NOTE: The item list is in a random order currently

## Related Issue

#4 

## Acceptance Criteria

- User is presented with a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again (Soon, Kind of Soon, Not Soon)
- When the user submits the form, the item is saved to the database, associated with the user’s token
- Along with the item name, an integer corresponding to the estimated number of days until next purchase is saved: 7 for “Soon”, 14 for “Kind of Soon”, and 30 for “Not Soon”
- The last purchased date should be set to null initially (i.e. you can create an item without purchasing it)
- Item names should be displayed as a simple, unordered list on the list view after they’re added

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

image

Screen Shot 2021-07-18 at 10 32 54 PM

### After

add-item-form

item-list

## Testing Steps / QA Criteria

- Pull down `sd-ia-addnewitem` branch
- `npm start`

### CODE REVIEW

**Review by ksiman14 — COMMENTED** [review]
This looks great! It should be easy to integrate with the work that Hulya and I did, we'll just have to make simple changes to lines 14-15 when we merge. 

I have one thing for thought - right now, if a user adds an item without selecting a frequency option, the frequency is set to 0. I'm wondering if this might present some challenges in the future. Should we refactor it so that there is a default selection (and frequency) that is 7, 14, or 30 days? 

I also cleared some of the old test items from the Firestore so it's easier to see when things are added in the new format.

**Comment by meganesu** [comment]
> Should we refactor it so that there is a default selection (and frequency) that is 7, 14, or 30 days?

Following up on this, plus @isaabutaa's suggestion on Slack: "If instead of giving it a default value of 0 or 7, maybe we can have a null default value if it's just a one time purchase. And then if they select another value, it'll swap out the default value."

To start with the simplest solution, I'd suggest making the field required (either by picking one of the existing values as a default for now, or preselecting one of the radio buttons when the form loads). And then you can create an additional issue to improve that behavior later on, if we have time. (That issue will also be a good place to track conversation about possible approaches!)

Great collaboration, team! 💪 🥳

File: src/ItemsList.js [hunk 1]
```diff
@@ -12,10 +12,12 @@ const ItemsList = () => {
       {error && <>Error</>}
       {snapshot && (
         <>
-          Collection:
-          {snapshot.docs.map((doc, index) => (
-            <pre key={index}>{JSON.stringify(doc.data())}</pre>
-          ))}
+          <h1>Collection:</h1>
```
Comment (meganesu) [inline src/ItemsList.js:15]: Way to add in semantic HTML! 🥳 A great way to improve accessibility. Nice job!

File: src/AddItem.js [hunk 2]
```diff
@@ -1,21 +1,91 @@
...
+    } else if (e.target.name === 'frequency') {
+      setFrequency(e.target.value);
+    }
+  };
+
+  return (
+    <div>
+      <h1>Smart Shopping List</h1>
+      <form onSubmit={handleClick}>
+        <label htmlFor="item">
```
Comment (meganesu) [inline src/AddItem.js]: I don't think your labels are working as expected here. The `for` attribute on your `label` element looks for a matching `id`, which at the moment doesn't exist on your `input` elements.

You can try playing around with the sample code on the [MDN `<label>` docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label). If the labels are connected correctly, you should be able to click on the label text and focus will move to the input. Try deleting the `id` from one of the inputs, and you'll see that clicking on that label no longer affects the checkbox.

(There are also some accessibility implications, which means unconnected labels will sound funny with a screen reader, but we'll talk more about that in the learning module in a few weeks!)

**Review by meganesu — CHANGES_REQUESTED** [review]
Looking great! Left one comment about a small labeling problem. Otherwise, this looks solid! Well done!

**Comment by sandaiiyahh** [comment]
As I was working on adding the id's, I realized we should also add an error handler for when a user submits an empty item input, which we can definitely implement in the future!

**Comment by isaabutaa** [comment]
> As I was working on adding the id's, I realized we should also add an error handler for when a user submits an empty item input, which we can definitely implement in the future!

Good idea, Sandy!

**Review by meganesu — APPROVED** [review]
🥳

**Pull request merged**